Wrappers/Python/ccpi/framework/framework.py

```diff
@@ -920,15 +920,34 @@ def __init__(self,
                     if key == 'spacing' :
                         self.spacing = value
                         
-        def subset(self, dimensions=None, **kw):
-            # FIXME: this is clearly not rigth
-            # it should be something like 
-            # out = DataContainer.subset(self, dimensions, **kw)
-            # followed by regeneration of the proper geometry. 
-            out = super(ImageData, self).subset(dimensions, **kw)
-            #out.geometry = self.recalculate_geometry(dimensions , **kw)
-            out.geometry = self.geometry
-            return out
+    def subset(self, dimensions=None, **kw):
+        # FIXME: this is clearly not rigth
+        # it should be something like 
+        out = DataContainer.subset(self, dimensions, **kw)
+        # followed by regeneration of the proper geometry. 
+        #out = super(ImageData, self).subset(dimensions, **kw)
+        #out.geometry = self.recalculate_geometry(dimensions , **kw)
+        
+        if out.number_of_dimensions > 1:
+        
+            out.geometry = ImageGeometry()
```

Comment:
This will create a default order of the axis labels. But we want to inherit from `self.geometry`
So I would rather

```python
out.geometry = self.geometry.copy()
```

This should fix, right?

Reply:
I do not think this is a correct way to fix this. If you simply copy the geometry, then you will have information about all dimensions, but you want only subset of them. I have an idea, let me check.

Reply:
will this
`out.geometry = ImageGeometry(dimension_labels = out.dimension_labels)`
solve the issue?

Reply:
Actually, this [call](https://github.com/vais-ral/CCPi-Framework/blob/fix_subset/Wrappers/Python/ccpi/framework/framework.py#L510) sets the `dimension_labels` in the appropriate order.

Reply:
I'm a bit confused now. `subset ` method of  `DataContainer` returns `dimension_labels`  in a correct order, I just need to pass them to `ImageGeometry`. Will my suggestion above work?